src/simcore_service_deployment_agent/cmd_utils.py

```diff
@@ -1,13 +1,44 @@
 import asyncio
 import logging
-from typing import List
+from asyncio.subprocess import Process
+from typing import Optional, Union
 
 from .exceptions import CmdLineError
 
 log = logging.getLogger(__name__)
 
 
-async def run_cmd_line(cmd: List[str], cwd_: str = ".") -> str:
+async def _wait_and_process_results(
+    process: Process, *, command: Union[str, list[str]], strip_endline: bool = True
+) -> Optional[str]:
+    # waits
+    stdout, stderr = await process.communicate()
+    log.debug("[{%s}] exited with %s]", command, process.returncode)
+
+    # process results
+    if process.returncode > 0:
+        error_data = ""
+        if stderr:
+            error_data = stderr.decode()
+            log.debug("\n[stderr]%s", error_data)
+        raise CmdLineError(command, error_data)
+
+    if stdout:
+        standard_output = stdout.decode()
+        log.debug("\n[stdout]%s", standard_output)
+
+        return standard_output.strip("\n") if strip_endline else standard_output
+    return None
+
+
+async def run_cmd_line(
```

Comment:
Minor: Commands like this, called some variation of `run_cmd` are found throughout the repos code (including testing files) and there is some duplication. Ideally, we should only use one robust version of this function for all code. Maybe this is the right moment to quickly go through, purge duplication and use this version of `run_cmd_line` everywhere?

Reply:
I would like to do this in a separate PR. In the best case, it is just a change but in the worst it can lead to a lot of refactoring.